webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -574,6 +603,94 @@ export class EditorControlService implements ZLUX.IEditor, ZLUX.IEditorMultiBuff
+    if (editor) {
+      contents = editor.getValue();
+    }
+    if (context != null) {
+      _activeDataset = context;
+    } else {
+      _activeDataset = _openDataset.filter(dataset => dataset.active === true)[0];
+    }
+    const model = _activeDataset.model;
+    const isDataset = _activeDataset.model.isDataset;
+    const fullName = isDataset ? _activeDataset.model.fileName : _activeDataset.model.name
+    if (!isDataset) {
+      this.snackBar.open(`${fullName} is not a dataset, invalid save route`, "Close", { duration: MessageDuration.Medium, panelClass: 'center' });
+      return;
+    }
+    if (!editor || contents === undefined) {
+      this.snackBar.open(`Cannot get editor or no contents to save`, "Close", {duration: MessageDuration.Medium, panelClass: 'center'});
+      return;
+    }
+
+    if (!destinationOverride && fullName) {
+
+      contents = editor.getValue().split('\n');
+      const requestUrl = ZoweZLUX.uriBroker.datasetContentsUri(fullName);
+      this.log.debug(`Should save contents to dataset. dataset=${fullName}, route=${requestUrl}`);
+      let result = isContentValidForDataset(contents, model.datasetAttrs);
+      if (result === true) {
+        this.saveDataset(context, _activeDataset, forceWrite);  
+      } else {
+        this.snackBar.open(`Content invalid for saving to dataset. Reason=${result}`, 'Close', { duration:MessageDuration.Long, panelClass: 'center'});
+      }
+    } else {
+      //dataset is new or needs some alteration we can't do yet
+      this.snackBar.open(`${_activeDataset.name} could not be saved, feature not yet implemented.`, 
+                         'Close', { duration: MessageDuration.Long,   panelClass: 'center' });    
+    }
+  }
+
+  saveDataset(context: ProjectContext, activeDataset: ProjectContext, forceWrite: Boolean): void {
+    const editor = this._editor.getValue();
+    let contents;
+    if(editor) {
+      contents = editor.getValue();
+    }
+    const model = activeDataset.model;
+    const isDataset = model.isDataset;
+    const fullName = isDataset ? model.fileName : model.name;
+    const etag = model.etag;
+    let headers = new Headers({'Etag': etag});
+    contents = editor.getValue().split('\n');
+    const requestUrl = ZoweZLUX.uriBroker.datasetContentsUri(fullName);
+    this.ngHttp.post(requestUrl, {records:contents, etag:etag}, {headers: headers, params:{force: forceWrite}}).subscribe(r => {
+      this.snackBar.open(`${activeDataset.name} has been saved!`,'Close', {duration:MessageDuration.Short, panelClass: 'center'});
+      /* Send buffer saved event */
+      this.bufferSaved.next({ buffer: activeDataset.model.contents, file: activeDataset.model.name });
+      this.openFileList.getValue()
+        .map(file => {
+          if (file.id === context.id) {
+            file.changed = false;
+          }
+          return file;
+        });
+    }, e => {
+      const err = e.json()
+      if(err) {
+        if(err.error.includes('etag mismatch')) {
```

Comment:
Add a TODO here because this message will almost certainly change from the message Tim wrote in the mock ZSS server